### PR TITLE
Move from Jest to uvu

### PR DESCRIPTION
**Pull request by kimoofey**
Move from Jest to uvu #1661

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,8 @@
     "transpiler"
   ],
   "scripts": {
-    "test": "jest --coverage && eslint . && ./test/version.js && check-dts && ./test/integration.js && size-limit"
+    "test": "c8 yarn unit && eslint . && ./test/version.js && check-dts && ./test/integration.js && size-limit",
```
Comment (ai) [inline package.json:56]: We need to copy `c8` config with coverage limit

https://github.com/ai/nanodelay/blob/main/package.json#L51-L57

File: test/browser.test.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
```
Comment (ai) [inline test/browser.test.ts:1]: We can remove this test. It tests rare case.

**Comment by ai** [comment]
We also need to change CI task to `yarn unit` https://github.com/postcss/postcss/blob/main/.github/workflows/test.yml#L63 and https://github.com/postcss/postcss/blob/main/.github/workflows/test.yml#L44

**Comment by ai** [comment]
We need to fix:

1. Coverage
2. Windows support (should we add `uvu` issue?)
3. Node.js 10 support, when there is not ESM. I am OK to create some `simple-test.js` script which will parse simple CSS if ESM support is not available (but maybe we can hack Node.js with custom loader?).

**Comment by kimoofey** [comment]
1. Do we need to increase coverage with tests or reduce this number in c8 rule in `package.json` (I've seen uncovered code and all of them with precommented line `// istanbul ignore next`)?
2. As a alternative, to rewrite `yarn unit` command to `uvu -r tsm test` (it will works on both platforms, but need to replace `version.js` and `integration.js`, so they don't run twice on `yarn test`)

**Comment by ai** [comment]
> I've seen uncovered code and all of them with precommented line `// istanbul ignore next`

Seems like we forget to convert `istanbul` ignore comments to `c8` ignore comments

> As a alternative, to rewrite yarn unit command to `uvu -r tsm test`

How will it help?

We already has `uvu -r tsm test` in `yarn unit` script. As I understand, the problem is with shell script.

**Comment by kimoofey** [comment]
> How will it help?

As in uvu [docs](https://github.com/lukeed/uvu/blob/master/docs/cli.md#cli), we can specify only target folder (without matching pattern, that different on some platforms).

**Comment by ai** [comment]
> As in uvu docs, we can specify only target folder (without matching pattern, that different on some platforms).

I afraid it will run all JS files. It is not our case. We need to find a way to run only specific files.

Solution:

1. Use a different command for Windows
2. Ask `uvu` team for some cross-platform API

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -60,6 +60,6 @@ jobs:
       - name: Install dependencies
         run: yarn install --frozen-lockfile
       - name: Run unit tests
-        run: npx jest
+        run: yarn unit:win
```
Comment (ai) [inline .github/workflows/test.yml:63]: If we need a script only in the single place, let’s put command here directly.

```suggestion
        run: npx uvu -r tsm test ^.*test.([tj]s)
```

**Comment by ai** [comment]
@kimoofey sorry, you will need to update PR because I merged another PR with test changes

**Comment by ai** [comment]
@kimoofey all other 8.3 PRs was merged. I hope we will not have more conflicts.

**Comment by kimoofey** [comment]
Everything is fine, except Node 10 support
I was looking for additional loader (for example this [one ](https://github.com/lukeed/loadr)), but still have not success with it

**Comment by ai** [comment]
> Everything is fine

We also have Windows issue.

> except Node 10 support

I asked `uvu` author https://github.com/lukeed/uvu/issues/162

If there is no way we can write a simple smoke test. A simple Node.js script (without any test framework) which will parse CSS and accept some basic plugin.

**Comment by ai** [comment]
@kimoofey `uvu`’s author recommends `uvu -r esm tests` to run on Node.js 10

**Comment by ai** [comment]
Great. Now only coverage is a blocker.

File: package.json [hunk 4]
```diff
@@ -54,7 +54,8 @@
     "transpiler"
   ],
   "scripts": {
-    "test": "jest --coverage && eslint . && node ./test/version.js && check-dts && node ./test/integration.js && size-limit"
+    "test": "c8 yarn unit && eslint . && node ./test/version.js && check-dts && node ./test/integration.js && size-limit",
+    "unit": "ts-node -T node_modules/uvu/bin.js -r esm test '\\.test\\.(ts|js)$'"
```
Comment (43081j) [inline package.json:58]: you can probably:

```sh
uvu -r ts-node/regiser -r esm test '\\.test\\.(ts|js)$'
```

Comment (kimoofey) [reply package.json:58]: you right, `ts-node` will be enough here. will remove `esm`

**Pull request merged**